src/main/java/com/teragrep/aer_01/EventContextConsumer.java

```diff
@@ -95,8 +127,50 @@ private String getRealHostName() {
         return hostname;
     }
 
+    private void startMetrics() {
+        this.jmxReporter.start();
+        this.slf4jReporter.start(1, TimeUnit.MINUTES);
+
+        // prometheus-exporter
+        CollectorRegistry.defaultRegistry.register(new DropwizardExports(metricRegistry));
+
+        ServletContextHandler context = new ServletContextHandler();
+        context.setContextPath("/");
+        jettyServer.setHandler(context);
+
+        MetricsServlet metricsServlet = new MetricsServlet();
+        ServletHolder servletHolder = new ServletHolder(metricsServlet);
+        context.addServlet(servletHolder, "/metrics");
+
+        // Start the webserver.
+        try {
+            jettyServer.start();
+        }
+        catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
     @Override
     public void accept(EventContext eventContext) {
+        int messageLength = eventContext.getEventData().getBody().length;
+        String partitionId = eventContext.getPartitionContext().getPartitionId();
+
+        records++;
+        allSize = allSize + messageLength;
+
+        metricRegistry.gauge(name(EventContextConsumer.class, "latency-seconds", partitionId), () -> new Gauge<Long>() {
+            @Override
+            public Long getValue() {
+                return Instant.now().getEpochSecond() - eventContext.getEventData().getEnqueuedTime().getEpochSecond();
+            }
+        });
+        metricRegistry.gauge(name(EventContextConsumer.class, "depth-bytes", partitionId), () -> new Gauge<Long>() {
+            @Override
+            public Long getValue() {
+                return eventContext.getLastEnqueuedEventProperties().getOffset() - eventContext.getEventData().getOffset();
+            }
+        });
```

Comment:
I am fairly sure that these gauges should not be recreated on every `accept()` call

Reply:
The gauge() function returns the gauge registered under the given name or creates a new one if one doesn't exist. So basically this will create a new gauge when a message is received from a not-seen-before partition, but most of the time it returns the correct gauge for the partition that the message is from.

Reply:
Ah getOrCreate method, thats fine